### PR TITLE
Update Xcode project and fix warnings

### PR DESCRIPTION
**Pull request by jordyd**

### CODE REVIEW

**Comment by qmathe** [comment]
Thanks a lot for the pull request! Looks good!

For the int casts,  I think it's better to do the casting in UKIntsEqual() macro, rather than have to cast the numbers almost every time UKIntsEqual() is used.

 So I'm going to update UKIntsEqual() macro in UnitKit and merge your changes.

File: Tests/TestCollectionTrait.m [hunk 1]
```diff
@@ -96,7 +96,7 @@ - (void) checkFixedCollectionCharacteristics
 - (void) testEmptyCollection
 {
     [self checkFixedCollectionCharacteristics];
-    UKIntsEqual(0, [self count]);
+    UKIntsEqual(0, (int)[self count]);
```
Comment (qmathe) [inline Tests/TestCollectionTrait.m]: Github doesn't let me choose select which changes I want to merge from your pull request.

So could you remove the int casts you added? I fixed the problem in UnitKit master.

File: Source/ETReflection.m [hunk 2]
```diff
@@ -240,7 +240,7 @@ BOOL ETGetInstanceVariableValueForKey(id object, id *value, NSString *key)
     assert(key != nil);
 
     const char *baseName = [key UTF8String];
-    int baseLength = strlen(baseName);
+    unsigned long baseLength = strlen(baseName);
```
Comment (ericwa) [inline Source/ETReflection.m]: `strlen` returns `size_t` so might as well use that here instead of `unsigned long`

File: Source/ETException.m [hunk 3]
```diff
@@ -79,7 +79,7 @@ - (void) raise
                 GLOBAL_EXCEPTION_STATE = EXCEPTION_ABORT;
         }
     }
-    nsexception_raise(self, _cmd);
+    nsexception_raise();
```
Comment (ericwa) [inline Source/ETException.m]: This change is wrong, I think.. it's calling an ObjC method via the IMP pointer, you need to pass the receiver and selector

Comment (jordyd) [inline Source/ETException.m]: Yeah that's what I thought but Xcode refused to compile it the other way, saying it required 0 parameters. Unfortunately I don't have enough knowledge of the codebase to really know the cause, but I was hoping that the tests covered it lol.

**Comment by ericwa** [comment]
Thanks, I merged this manually with a few tweaks

**Pull request merged**